Helper/PaymentMethods.php

```diff
@@ -165,95 +163,119 @@ protected function fetchAlternativeMethods($country)
-                    // Explicitly setting theme
-                    $themeCode = "Magento/blank";
-
-                    $themeId = $this->design->getConfigurationDesignTheme(\Magento\Framework\App\Area::AREA_FRONTEND);
-                    if (!empty($themeId)) {
-                        $theme = $this->themeProvider->getThemeById($themeId);
-                        if ($theme && !empty($theme->getCode())) {
-                            $themeCode = $theme->getCode();
-                        }
-                    }
-
-                    $params = [];
-                    $params = array_merge(
-                        [
-                            'area' => \Magento\Framework\App\Area::AREA_FRONTEND,
-                            '_secure' => $this->request->isSecure(),
-                            'theme' => $themeCode
-                        ],
-                        $params
-                    );
-
-                    $asset = $this->assetRepo->createAsset(
-                        'Adyen_Payment::images/logos/' .
-                        $paymentMethodCode . '.png',
-                        $params
-                    );
-
-                    $placeholder = $this->assetSource->findSource($asset);
-
-                    $icon = null;
-                    if ($placeholder) {
-                        list($width, $height) = getimagesize($asset->getSourceFile());
-                        $icon = [
-                            'url' => $asset->getUrl(),
-                            'width' => $width,
-                            'height' => $height
-                        ];
-                    }
-                    $paymentMethod['icon'] = $icon;
+
+                $asset = $this->assetRepo->createAsset(
+                    'Adyen_Payment::images/logos/' .
+                    $paymentMethodCode . '.png',
+                    $params
+                );
+
+                $placeholder = $this->assetSource->findSource($asset);
+
+                if ($placeholder) {
+                    list($width, $height) = getimagesize($asset->getSourceFile());
+                    $icon = [
+                        'url' => $asset->getUrl(),
+                        'width' => $width,
+                        'height' => $height
+                    ];
+                } else {
+                    $icon = [
+                        'url' => 'https://checkoutshopper-live.adyen.com/checkoutshopper/images/logos/medium/' .$paymentMethodCode . '.png',
+                        'width' => 77,
+                        'height' => 50
+                    ];
                 }
-                $paymentMethods[$paymentMethodCode] = $paymentMethod;
+
+                $paymentMethodsExtraDetails[$paymentMethodCode]['icon'] = $icon;
+
+                //todo check if it is needed
+                // check if payment method is an open invoice method
+                $paymentMethodsExtraDetails[$paymentMethodCode]['isOpenInvoice'] =
```

Comment:
keep it here until we get to the frontend maybe we won't need it after refactoring the frontend